include/deal.II/grid/tria_object.h

```diff
@@ -186,87 +160,51 @@ namespace internal
 
     //----------------------------------------------------------------------//
 
-    template <int structdim>
-    inline TriaObject<structdim>::TriaObject()
-    {
-      for (const unsigned int i : GeometryInfo<structdim>::face_indices())
-        faces[i] = -1;
-    }
-
 
-    template <int structdim>
-    inline TriaObject<structdim>::TriaObject(const int i0, const int i1)
+    inline TriaObject::TriaObject(std::initializer_list<int> faces)
     {
-      Assert(structdim == 1, ExcImpossibleInDim(structdim));
-      faces[0] = i0;
-      faces[1] = i1;
+      this->faces = faces;
     }
 
 
-    template <int structdim>
-    inline TriaObject<structdim>::TriaObject(const int i0,
-                                             const int i1,
-                                             const int i2,
-                                             const int i3)
-    {
-      Assert(structdim == 2, ExcImpossibleInDim(structdim));
-      faces[0] = i0;
-      faces[1] = i1;
-      faces[2] = i2;
-      faces[3] = i3;
-    }
 
-
-    template <int structdim>
-    inline TriaObject<structdim>::TriaObject(const int i0,
-                                             const int i1,
-                                             const int i2,
-                                             const int i3,
-                                             const int i4,
-                                             const int i5)
+    inline TriaObject::TriaObject(std::initializer_list<unsigned int> faces)
     {
-      Assert(structdim == 3, ExcImpossibleInDim(structdim));
-      faces[0] = i0;
-      faces[1] = i1;
-      faces[2] = i2;
-      faces[3] = i3;
-      faces[4] = i4;
-      faces[5] = i5;
+      this->faces.reserve(faces.size());
+      for (const auto face : faces)
+        this->faces.push_back(face);
```

Comment:
This doesn't do what you want. It assumes that the object is empty beforehand. I think what you want is
```
  this->faces = std::vector<...>(faces);
```
Extra points if you name the argument differently than the member variable :-)

Reply:
Actually, it does. It's the constructor; so it is empty at that stage. The code you are suggesting does not work since the types are different (`int` vs. `unsigned int`). The inconsistent usage of `int` and `unsigned int` in `Triangulation`  is a larger issue (see #10431).